File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/director_wstring_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

director_wstring


B=@(string) subclass(A(string),\
		     'get_first',A.get_first(self) + " world!",\
		     'process_text',@(self) self.smem = u"hello"\
		     );

b = B("hello");

b.get(0);
if (!strcmp(b.get_first(),"hello world!"))
  error(b.get_first())
endif

b.call_process_func();

if (!strcmp(b.smem,"hello"))
  error(smem)
endif
  

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/director_wstring_runme.py
from director_wstring import *


class B(A):

    def __init__(self, string):
        A.__init__(self, string)

    def get_first(self):
        return A.get_first(self) + u" world!"

    def process_text(self, string):
        self.smem = u"hello"


b = B(u"hello")

b.get(0)
if b.get_first() != u"hello world!":
    print b.get_first()
    raise RuntimeError


b.call_process_func()

if b.smem != u"hello":
    print smem
    raise RuntimeError
